sbi/utils/get_nn_models.py

```diff
@@ -345,3 +350,35 @@ def classifier_nn(
         raise ValueError(f"'model' must be one of ['linear', 'mlp', 'resnet'].")
 
     return neural_net
+
+
+def _check_1d_flow_limitations(
+    model: str, output_numel: int, parameter_or_data: str = ""
+):
+    """
+    Gives a warning or error due to limitations of flows when estimating 1D densities.
+
+    For MAFs this gives a warning if the learned density is 1D, where MAFs are
+        inherently limited to Gaussian densities. For NSFs this raises an error if the
+        learned density is 1D. This is due to implementation issues.
+
+    Args:
+        model: The used density estimator, one of maf / mdn / made / nsf.
+        output_numel: Dimensionality of the space output space of the density estimator.
+        parameter_or_data: Whether the output of the density estimator is the parameter
+            space theta (which is the case for SNPE) or the data space x (which is the
+            case for SNL). Is used to add this information to the message raised by this
+            function.
+    """
+
+    if output_numel == 1 and model == "maf":
+        warn(
+            "In one-dimensional " + parameter_or_data + " spaces, MAFs are limited to a"
+            ' Gaussian posterior. Consider setting density_estimator="mdn" instead.'
```

Comment:
It's a good idea to use backticks around really functioning code, e.g. `density_estimator="mdn"`.